DirectProgramming/DPC++FPGA/Tutorials/Features/ac_int/README.md

```diff
@@ -0,0 +1,250 @@
+
+Please refer to the [oneAPI DPC++ FPGA Optimization Guide](https://software.intel.com/content/www/us/en/develop/documentation/oneapi-fpga-optimization-guide/top/optimize-your-design/resource-use/data-types-and-operations/var-prec-fp-sup/adv-disadv-ac-dt.html) to see advantages and limitations of `ac_int` data types.
+
+### Simple Code Example
+
+An `ac_int` number can be defined as follows:
+```cpp
+ac_int<W, S> a;
+```
+Here `W` is the width in bits and `S` is a bool indicating if the number is signed. Signed numbers use the most significant bit (MSB) to store the sign bit.
+
+To use the `ac_int` type in your code, you must include the following header:
+
+```cpp
+#include <sycl/ext/intel/ac_types/ac_int.hpp>
+```
+Additionally, you must pass the flag `-qactypes` on Linux or `/Qactypes` on Windows to the `dpcpp` command when compiling your SYCL program in order to ensure that the headers are correctly included. In this tutorial, this is done in `src/CMakeLists.txt`.
+
+### Basic Operations and Promotion Rules
+
+When using `ac_int`, the results of addition, subtraction, multiplication, and division operations are automatically promoted to the number of bits needed to represent all possible results without overflowing. However, the data type you use to store the result may result in truncation.
+
+For example, the addition of two 8-bit integers results in a 9-bit result to support overflow. Internally, the result will be 9-bit. However, if the user attempts to store the result in an 8-bit container, `ac_int` will let the user do this, which leads to the most significant bit being discarded. The responsibility lies on the user to use the correct data type.
+
+These promotion rules are consistent across all architectures, so the behavior will be equivalent on x86 or on FPGA.
+
+### Shift Operations
+
+The behavior of shift operations of `ac_int` data types is slightly different from shift operations of native integer types. For full details, see the Algorithmic C (AC) Datatypes documentation available at https://hlslibs.org/. Some key points to remember are as follows:
+  - If the data type of the shift amount is not explicitly `unsigned` (either using `ac_int<N, *false*>` or using the `unsigned` keyword), then the compiler will generate a more complex shifter that allows negative shifts and positive shifts. A shift by a negative amount is equivalent to a positive shift in the opposite direction. Normally, you will not want to use negative shifting, so you should use an `unsigned` data type for the shift value to obtain a more resource efficient shifter.
```

Comment:
\*false\* does not render as italics in the GitHub markdown viewer, presumably because it's inside back-ticks.  I recommend removing the * notation

Reply:
Will do.